magenta/models/music_vae/music_vae_export.py

```diff
@@ -0,0 +1,219 @@
+        else:
+            z_input = None
+
+        if vae_config.data_converter.control_depth > 0:
+            c_input = tf.placeholder(tf.float32, shape=[None, vae_config.data_converter.control_depth])
+        else:
+            c_input = None
+
+        inputs = tf.placeholder(
+            tf.float32,
+            shape=[batch_size, None, vae_config.data_converter.input_depth])
+        controls = tf.placeholder(
+            tf.float32,
+            shape=[batch_size, None, vae_config.data_converter.control_depth])
+        inputs_length = tf.placeholder(
+            tf.int32,
+            shape=[batch_size] + list(vae_config.data_converter.length_shape))
+
+        max_length = tf.placeholder(tf.int32, shape=())
+        input_tensors = InputTensors(inputs, controls, inputs_length, c_input, z_input, temperature, max_length)
+
+        # used in the trained_model decode functions, run in the stored session
+        OutputTensors = namedtuple(
+            "VAEOutputTensors",
+            ["outputs", "decoder_results", "mu", "sigma", "z"]
+        )
+        outputs, decoder_results = model.sample(
+            batch_size,
+            max_length=max_length,
+            z=z_input,
+            c_input=c_input,
+            temperature=temperature,
+            **sample_kwargs)
+
+        if vae_config.hparams.z_size:
+            q_z = model.encode(inputs, inputs_length, controls)
+            mu = q_z.loc
+            sigma = q_z.scale.diag
+            z = q_z.sample()
+        output_tensors = OutputTensors(outputs, decoder_results, mu, sigma, z)
+
+        signature_defs = build_signature_defs(input_tensors, output_tensors)
+
+    return vae_graph, signature_defs
+
+
+# This function has too many arguments, pass a data structure later
+def build_signature_defs(input_tensors, output_tensors):
+    sample_signature_def = build_sample_signature(input_tensors, output_tensors)
+    encode_signature_def = build_encode_signature(input_tensors, output_tensors)
+    decode_signature_def = build_decode_signature(input_tensors, output_tensors)
+
+    signature_def_map = {
+        'sample': sample_signature_def,
+        'encode': encode_signature_def,
+        'decode': decode_signature_def
+    }
+
+    return signature_def_map
+
+
+# TODO create a named registry of signature def builders keyed by name for better type safety
+# Sample seems almost exactly like decode from the trained model code, possibly redundant
+def build_sample_signature(input_tensors, output_tensors):
+    sample_inputs = {
+        'temperature': saved_model.utils.build_tensor_info(input_tensors.temperature),
+        'max_length': saved_model.utils.build_tensor_info(input_tensors.max_length)
+    }
+    if input_tensors.c_input is not None:
+        sample_inputs['c_input'] = saved_model.utils.build_tensor_info(input_tensors.c_input)
+    if input_tensors.z_input is not None:
```

Comment:
as mentioned above, you should feed in a random normal sample for the z

Reply:
See above